rules.py

```diff
@@ -953,25 +953,30 @@ def check(self, element):
         object_id = element.get("objectId")
         if element.tag == "Candidate":
             ballot_name = element.find("BallotName")
-            if ballot_name.find("Text") is not None:
-                name = ballot_name.find("Text").text
-                if name is not None and name == name.upper():
```

Comment:
IMHO, this shall be a check for none empty string, you can use not name.strip() instead (of name is not None)

Reply:
Updated the conditional checks and covered the corner cases.